chainerx_cc/chainerx/cuda/memory_pool.h

```diff
@@ -74,10 +117,17 @@ class MemoryPool {
 private:
     friend class cuda_internal::MemoryPoolTest;  // for unit-tests
 
+    // Rounds up the memory size to fit memory alignment of cudaMalloc.
+    size_t GetAllocationSize(size_t bytesize) { return ((bytesize + kAllocationUnitSize - 1) / kAllocationUnitSize) * kAllocationUnitSize; }
```

Comment:
This function prevents `free_bins_` being too huge size.

Reply:
Yes, I understand that. But it does not sound related to CUDA.

Reply:
CUDA aligns to multiples of 512 (when Iimplemented BFC in CuPy, so maybe CUDA8?).
You can see it with cudaMalloc(1) twice and prints the address of ptrs.

But, it might be changed in the future, say, to multiples of 1024.
kAllocationUnitSize should be changed to 1024 in such future.
Better to change this comment to describe this fact.

Reply:
Of course, increasing kAllocationUnitSize more is another choice, but at least, decreasing kAllocationUnitSize to 256 or less is nonsense because of the fact.

Reply:
I think that it might be better to define two constant values: "unit size of CUDA malloc" and "unit size of MemoryPool malloc".

Reply:
It would be nice, but "unit size of CUDA malloc" is not used in codes actually.
Writing it in comments would be enough.

Reply:
BTW, I discovered my gist https://gist.github.com/sonots/41daaa6432b1c8b27ef782cd14064269

Reply:
It seems it's dependent on the underlying allocator functions.
In my environment, `cudaMallocManaged` gives the allocation unit size of 4096.
Perhaps it's better to make it determined at runtime, instead of a constant.


Maybe current implementation is fine, but could you at least add TODO?

>I think that it might be better to define two constant values: "unit size of CUDA malloc" and "unit size of MemoryPool malloc".

Write about this, too.